app/schema/questionnaire.py

```diff
@@ -43,16 +43,22 @@ def register(self, item):
             raise QuestionnaireException('{} is a duplicate id'.format(item.id))
 
         self.items_by_id[item.id] = item
-        self._register_alias(item)
 
         # Build the two-way relationship
         item.questionnaire = self
 
+    def register_aliases(self):
+        for _, item in self.items_by_id.items():
+            self._register_alias(item)
+
     def _register_alias(self, item):
         # Register the alias if the item has one
         if isinstance(item, Answer) and item.alias is not None:
             if item.alias not in self.aliases.keys():
-                self.aliases[item.alias] = item.id
+                self.aliases[item.alias] = {
+                    "item_id": item.id,
```

Comment:
Yeah it's an answer_id. As far as I know, we only use aliases on Answers right now, so I'll rename this to answer_id for clarity.